### DISCUSSION TITLE
[Navy BlockCache] Is device resize handled without data loss?

### DISCUSSION THREAD

**Discussion by dannycho7** [facebook/CacheLib#242] [Q&A] [answered]
For Navy BlockCache, I noticed that [`ItemRecord`](https://github.com/facebook/CacheLib/blob/main/cachelib/navy/block_cache/Index.h#L59) addresses are aligned by an amount that is [dynamically calculated based on the device size](https://github.com/facebook/CacheLib/blob/main/cachelib/navy/block_cache/BlockCache.cpp#L92-L105), with a minimum alignment of 512 bytes.

Is it possible for a device upsize (e.g. from 1 TiB to 16 TiB) to cause data invalidation due to the alignment of addresses changing?

**Comment by haowu14** [comment, marked as answer]
Yes. And the `Device` class size is just the amount of space assigned to the cache. So just a config change without physical device change could cause the data to drop.

**Comment by pbhandar2** [comment]
Closing this discussion as it has been inactive. If this is still relevant, then please create a new discussion post and we will get back to you as soon as we can. Thank you!
